x/mev/module.go

```diff
@@ -0,0 +1,203 @@
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+	// Register your routes here if needed
+
+	//_ = types.RegisterQueryHandlerClient(context.Background(), mux, types.NewMsgClient(clientCtx))
+
+}
+
+// GetTxCmd returns the root tx command for the mev module.
+func (AppModuleBasic) GetTxCmd() *cobra.Command {
+	return nil
+	//return cli.GetTxCmd()
+}
+
+// GetQueryCmd returns the root query command for the mev module.
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return nil
+	//return cli.GetQueryCmd()
+}
+
+type AppModule struct {
+	AppModuleBasic
+	keeper *mev.Keeper
+}
+
+func NewAppModule(cdc codec.Codec, k *mev.Keeper) AppModule {
+	return AppModule{
+		AppModuleBasic: NewAppModuleBasic(cdc),
+		keeper:         k,
+	}
+}
+
+// Name returns the mev module's name.
+func (AppModule) Name() string {
+	return types.ModuleName
+}
+
+// RegisterInvariants registers the mev module invariants.
+func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
+
+func NewHandler(k *mev.Keeper) sdk.Handler {
+	//msgServer := keeper.NewMsgServerImpl(k)
+
+	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+
+		fmt.Println("MEV HANDLER MEV")
+
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+
+		switch msg := msg.(type) {
+		//case *types.MsgEVMTransaction:
+		//	res, err := msgServer.EVMTransaction(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		//case *types.MsgSend:
+		//	res, err := msgServer.Send(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		//case *types.MsgRegisterPointer:
+		//	res, err := msgServer.RegisterPointer(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		//case *types.MsgAssociateContractAddress:
+		//	res, err := msgServer.AssociateContractAddress(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		default:
+			errMsg := fmt.Sprintf("unrecognized %s message type MEV MEV MEV: %T", types.ModuleName, msg)
+			return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
+		}
+	}
+}
+
+// Route returns the message routing key for the mev module.
+func (am AppModule) Route() sdk.Route {
+	return sdk.NewRoute(types.RouterKey, NewHandler(am.keeper))
+}
+
+// QuerierRoute returns the mev module's querier route name.
+func (AppModule) QuerierRoute() string {
+	return types.QuerierRoute
+}
+
+// LegacyQuerierHandler returns the mev module sdk.Querier.
+func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
+	return nil
+}
+
+// InitGenesis performs genesis initialization for the mev module.
+func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
+	var genesisState types.GenesisState
+	cdc.MustUnmarshalJSON(data, &genesisState)
+	return []abci.ValidatorUpdate{}
```

Comment:
why return an empty slice rather than nil? 

does this actually have any side effects?

Reply:
Interestingly, other modules returns empty slice rather then `nil`. No idea why.
In theory, I think that if you had a type alias and declared method it would make a difference if you case `nil` slice and empty one, but I don't see anything like this going on here.